AlfrescoApp/Model/Managers/RealmSyncManager+CoreDataMigration.m

```diff
@@ -32,6 +32,103 @@
 
+- (BOOL)isContentMigrationNeeded
+{
+    NSUserDefaults *defaults = [[NSUserDefaults alloc] initWithSuiteName:kAlfrescoMobileGroup];
+    BOOL isMigrationNeededResult = ![defaults objectForKey:kHasSyncedContentMigrationOccurred];
+    
+    return isMigrationNeededResult;
+}
+
+- (void)initiateContentMigrationProcess
+{
+    void (^saveContentMigrationOccured)() = ^()
+    {
+        NSUserDefaults *defaults = [[NSUserDefaults alloc] initWithSuiteName:kAlfrescoMobileGroup];
+        [defaults setObject:@YES forKey:kHasSyncedContentMigrationOccurred];
+        [defaults synchronize];
+    };
+    
+    if ([[AccountManager sharedManager] allAccounts].count)
+    {
+        __block BOOL migrationOfRealmFilesSucceded;
+        __block BOOL migrationOfSyncFolderSucceded;
+        
+        [self migrateRealmFilesWithCompletionBlock:^(BOOL succeeded, NSError *error) {
+            migrationOfRealmFilesSucceded = succeeded;
+        }];
+        
+        [self migrateSyncFolderWithCompletionBlock:^(BOOL succeeded, NSError *error) {
+            migrationOfSyncFolderSucceded = succeeded;
+        }];
+        
+        if (migrationOfRealmFilesSucceded && migrationOfSyncFolderSucceded)
+        {
+            saveContentMigrationOccured();
+        }
+    }
+    else
+    {
+        saveContentMigrationOccured();
+    }
+}
+
+- (void)migrateRealmFilesWithCompletionBlock:(AlfrescoBOOLCompletionBlock)completionBlock
+{
+    NSURL *sharedAppGroupFolderURL = [[NSFileManager defaultManager] containerURLForSecurityApplicationGroupIdentifier:kSharedAppGroupIdentifier];
+    NSString *documentsDirectoryPath = [NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES) lastObject];
+    NSURL *url = [NSURL URLWithString:documentsDirectoryPath];
+    NSError *error = nil;
+    NSArray *array = [[NSFileManager defaultManager] contentsOfDirectoryAtURL:url includingPropertiesForKeys:@[NSURLNameKey] options:0 error:&error];
+    
+    if (error)
+    {
+        AlfrescoLogError(@"Error fetching documents folder content. Error: %@", error.localizedDescription);
+    }
+
+    __block BOOL moveErrorOccured;
+    
+    [array enumerateObjectsUsingBlock:^(NSURL *url, NSUInteger idx, BOOL * _Nonnull stop) {
+        NSString *lastPathComponent = url.lastPathComponent;
+        
+        if ([lastPathComponent containsString:@"realm"])
+        {
+            NSString *destinationPath = [sharedAppGroupFolderURL.path stringByAppendingPathComponent:lastPathComponent];
+            NSError *moveError;
+            [[AlfrescoFileManager sharedManager] moveItemAtPath:url.path toPath:destinationPath error:&moveError];
+            
+            if (moveError)
+            {
+                AlfrescoLogError(@"Error moving a realm file. Error: %@", moveError.localizedDescription);
+                moveErrorOccured = YES;
+            }
+        }
+    }];
+    
+    completionBlock(moveErrorOccured, nil);
```

Comment:
Completion block should return any error that might arise during the move process, instead of using nil.